nntrainer/utils/ini_wrapper.cpp

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file ini_wrapper.cpp
+ * @date 08 April 2021
+ * @brief NNTrainer Ini Wrapper helps to save ini
+ * @note this is to be used with ini_interpreter
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <ini_wrapper.h>
+
+#include <nntrainer_error.h>
+#include <parse_util.h>
+#include <regex>
+
+namespace nntrainer {
+
+IniSection::IniSection(const std::string &name) : section_name(name) {}
+
+IniSection::IniSection(const std::string &section_name,
+                       const std::string &entry_str) :
+  IniSection(section_name) {
+  setEntry(entry_str);
+}
+
+IniSection::IniSection(IniSection &from, const std::string &section_name,
+                       const std::string &entry_str) :
+  IniSection(from) {
+  if (!section_name.empty()) {
+    this->section_name = section_name;
+  }
+  if (!entry_str.empty()) {
+    setEntry(entry_str);
+  }
+}
+
+void IniSection::print(std::ostream &out) {
+  out << '[' << section_name << ']' << std::endl;
+  for (auto &it : entry)
+    out << it.first << " = " << it.second << std::endl;
+}
+
+void IniSection::setEntry(const std::map<std::string, std::string> &entry) {
+  for (auto &it : entry) {
+    this->entry[it.first] = it.second;
+  }
+}
+
+void IniSection::setEntry(const std::string &entry_str) {
```

Comment:
it's ok for now, but later, you could consider to use std::vector<std::string> than using a separator `|`.

Reply:
yes, this could be optimized later, fyi) this method should kept for the test codes though.
I am putting a todo.

Reply:
When this is updated, the test codes can be updated as well (in a later PR).

Reply:
For sure :)